src/App.js

```diff
@@ -1,33 +1,48 @@
-import React, { useState } from 'react';
-import { UserContext } from './components/UserContext';
+import React from 'react';
 import MovieList from './components/MovieList';
 import BottomPanel from './components/BottomPanel';
 import Title from './components/Title';
+import Provider from './components/Provider'
```

Comment:
I liked the idea of moving the Provider logic out into its own Provider component. This has 2 positive side effects from what I can see.

1. It simplifies the logic taking place in the App components render function.
2. It lets it become a bit more reusable (Not that I think we would ever need to reuse it).